src/dags/importscripts/import_processenverbaalverkiezingen.py

```diff
@@ -0,0 +1,134 @@
+import csv
+import ftplib
+from more_ds.network.url import URL
+from os import path as ospath
+from typing import List, TypedDict, Tuple, Optional, Generator
+
+
+class Data(TypedDict):
+    """Datastructure processenverbaal verkiezingen"""
+
+    verkiezingsjaar: int
+    volgnummer: str
+    uri: str
+    documentnaam: str
+    stemlocatie: str
+    id: str
+
+
+class FTPListing:
+    """This class is used for traversing a given folder base on a FTP
+    server (objectstore) using BFS (Breadth-First search) algorithm.
+    """
+
+    def __init__(self, connection: ftplib.FTP) -> None:
+        """Initialize
+
+        Args:
+            Connection: Hold the connection to the objectstore
+
+        """
+        self.connection = connection
+
+    def list_dirs_and_files(self, _path: str) -> Tuple[List[str], Optional[List[str]]]:
+        """Return files and directories within a path
+        So it can be used to identifiy if there is need for another directory scan
+        to locate files. Files already found can be used to get the URI.
+
+        Args:
+            _path: The path for looking for files and directories (within the same path)
+
+        Returns:
+            list of found dictories and files
+
+        """
+
+        list_, dirs, files = [], [], []
+        try:
+            self.connection.cwd(_path)
+        except Exception:
+            return [], []
+        else:
+            self.connection.retrlines("LIST", lambda x: list_.append(x.split()))
+            for info in list_:
+                type, name = info[0], info[-1]
+                if type.startswith("d"):
+                    dirs.append(name)
+                else:
+                    files.append(name)
+            return dirs, files
+
+    def traverse_folder(self, path: str = "/") -> Generator:
```

Comment:
```python
-> Iterator[Tuple[str, List[str], List[str]]]
```
Though you might want to consider using a `dataclass` for this.  After all in the function `save_data` you are referring to `data[0]`, `data[1]` and `data[2]`. Without looking at the function body of `traverse_folder` it is not immediately clear what each element refers to. Hence you might want to consider this:

```python
from dataclasses import dataclass, field
from typing import List

@dataclass
class DirList:
    path: str
    dirs: List[str] = field(default_factory=list)
    files: List[str] = field(default_factory=list)
```
And use that in your `list_dirs_and_files` and traverse_folder` methods.

Anyway with `DirList` in place code in `save_data` would change from something like:

```python
    for data in get_listing.traverse_folder(startfolder):
        for file in data[2]:
```
to
```python
    for data in get_listing.traverse_folder(startfolder):
        for file in data.files:
```
See the difference between `data[2]` versus `data.files`?

In addition `data[0]` becomes `data.path` (and `data[1]` becomes `data.dirs`)

Reply:
:+1: interesting thanks! First time using the dataclass, it's very handy. I wonder if this could also work with a TypedDict. Making it duck typing proof if there is a use case for it off course, next time maybe!